### PR TITLE
feature-FEMR147-SearchByPhoneNumber

### PR DESCRIPTION
**Pull request by Thraxus**
Amended search message to include phone number as an option.  Major code changes were complete as a side effect of FEMR157.  Search is only implemented on home and triage. Medical and Pharmacy only run on patient ID.  It's worth noting that home uses a typeahead format, so patients can be filtered down in real time.  Triage requires an exact match else you'll be presented with a search results page where the user can choose the patient they are after.

### CODE REVIEW

**Comment by kevinzurek** [comment]
@Thraxus it looks like the most recent commit (b2a2e4a80e4eb165349577cd8b618f4f76d441d2) has a lot of the changes from FEMR157 which is making it hard to review. Perhaps a rebase would help?

**Comment by Thraxus** [comment]
@kevinzurek Yeah, I tried, and apparently failed again.  See, this is the area Git and I aren't on the same page.  I rebase, it complains.  I update, it complains.  I pull, push, it still complains.  Finally something works, but in the end either nothing looks different or I'm committing ever one's work with my push.

In any case, d5f5bfa are the only changes on this request.  5 text fields updated to reflect the new search method.

**Comment by kevinzurek** [comment]
@Thraxus a few common things that might be happening:

During a rebase you are replaying commits on top of updated code, this often causes merge conflicts. if you do `git status` when the rebase pauses it will tell you where you are at in the process and also if you are done. The red files displayed when you do `git status` require you to resolve merge conflicts. Manually resolve them in the file and then do `git add filename` then `git rebase continue`. _Do not commit_ during the rebase, this would cause complaining.

If your work is all in one commit, i like using `git cherry-pick` to make life easy:
1) `git checkout master`
2) `git pull upstream master`
3) `git branch feature-FEMR123-feature`
4) `git checkout feature-FEMR123-feature`
5) `git cherry-pick d5f5bfaf74dd670e96c70e00abc27c6230226879`

1 & 2 update your master branch
3 & 4 make a new branch off the most up to date master branch
5 moves the commit with your work to your new branch (the unique hash should match the hash of the commit in your feature-FEMR147-SearchByPhoneNumber branch)

**Comment by kevinzurek** [comment]
@Thraxus I picked out that commit, but am not able to pull up a patient by their phone number. please let me know if i'm missing something!! Thanks!

**Comment by Thraxus** [comment]
Checking on this now.

**Comment by Thraxus** [comment]
@kevinzurek Looks like what was working before was only working because it was keying off the ID field for the phone number.  The fixes I did for [FEMR-157](https://github.com/FEMR/femr/pull/560/commits/67b6c179331300b3d03d3b5c38510cb2b07a0bf6) broke the implementation here.  It's working properly now.  I'm going to attempt a better cleanup / push with your suggestions above.  

Stay tuned.

**Comment by kevinzurek** [comment]
It's working properly now - meaning you need to push up new code right? Trying to get this merged and on the servers :)

**Comment by Thraxus** [comment]
Yeah, just did a push.  Seeing how clean I made it this time.

**Comment by kevinzurek** [comment]
Try cherry picking:

`git checkout master`
`git pull upstream master`
`git branch feature-FEMR147-SearchByPhoneNumber2`
`git checkout feature-FEMR147-SearchByPhoneNumber2`
`git cherry-pick 80296e8ba09e1ccb5c181b93a2bf4cff6ce6ca8b`
`git cherry-pick d5f5bfaf74dd670e96c70e00abc27c6230226879`

File: app/femr/business/services/system/SearchService.java [hunk 1]
```diff
@@ -475,6 +481,20 @@ public SearchService(IRepository<IConceptDiagnosis> diagnosisRepository,
 
         List<? extends IPatient> patients;
 
+        //Build the Query
```
Comment (Thraxus) [inline app/femr/business/services/system/SearchService.java]: Somehow this block got duplicated.  Will fix and reupload.

Comment (kevinzurek) [inline app/femr/business/services/system/SearchService.java]: at which step?

**Comment by Thraxus** [comment]
Didn't cherry pick, but it looks clean to me.  Please review and let me know what you think.

**Comment by kevinzurek** [comment]
Looks better at first glance, will check out

**Comment by kevinzurek** [comment]
Very nice work, thank you for fixing that so quickly. Rebased and merged manually:

master: 4e2fc97bdb8c06fbc401a4b42bbc94e3fcfe76dc
release-2.3: e3640227285fa432bb991032a3412972e8860ec7

I'm thinking we should add 2 new change requests in JIRA:
1) Updating a patients phone number during a 2nd encounter (try making a patient then seeing them in Triage for a second encounter and trying to update phone #) I think this is blocked by FEMR-265 so it can't be completed right now.

2) Finding a solution for this nice little hack:
https://github.com/FEMR/femr/commit/e3640227285fa432bb991032a3412972e8860ec7#diff-b61c039a8e98d6f3afc064dd9f5480e4R466. I think this is related to FEMR-29.

If you agree with both please go ahead and create them in JIRA otherwise i'll get to it later :+1:

**Pull request closed without merging**